db/document.go

```diff
@@ -829,6 +837,48 @@ func (doc *Document) UpdateExpiry(expiry uint32) {
 
 //////// CHANNELS & ACCESS:
 
+func (doc *Document) updateChannelHistory(channelName string, seq uint64, addition bool) {
+	// Check if we already have an entry for
+	for idx, historyEntry := range doc.ChannelHistory {
+		if historyEntry.Name == channelName {
+			// If we are here there is an existing entry for this channel
+			// If addition we need to:
+			// - Move existing history entry to old channels
+			// - Remove existing entry from current channels
+			// - Add new entry with start seq
+			// If removal / not addition then we can simply add the end seq
+			if addition {
+				doc.OldChannelHistory = append(doc.OldChannelHistory, historyEntry)
+				doc.ChannelHistory = append(doc.ChannelHistory[:idx], doc.ChannelHistory[idx+1:]...)
+				doc.ChannelHistory = append(doc.ChannelHistory, DocumentChannelHistoryEntry{
+					Name:  channelName,
+					Start: seq,
+				})
+			} else {
+				doc.ChannelHistory[idx].End = seq
+			}
+			return
+		}
+	}
+
+	// If we get to this point there is no existing entry
+	// If addition we can just simply add it
+	// If its a removal / not addition then its a legacy document. Start seq is 1
```

Comment:
Can you add 'Start seq is 1 because we don't know when legacy channels were added'.